lib/libzfs/libzfs_changelist.c

```diff
@@ -541,15 +542,16 @@ change_one(zfs_handle_t *zhp, void *data)
 			uu_avl_insert(clp->cl_tree, cn, idx);
 		} else {
 			free(cn);
-			zfs_close(zhp);
 		}
 
-		if (!clp->cl_alldependents)
-			return (zfs_iter_children(zhp, change_one, data));
-	} else {
-		zfs_close(zhp);
+		if (!clp->cl_alldependents) {
+			ret = zfs_iter_children(zhp, change_one, data);
+			zfs_close(zhp);
+			return (ret);
```

Comment:
If you wanted to you could just initialize `ret` to 0 and not have the extra `zfs_close` and `return` here but rely on the `zfs_close` and `return` at the end of the function to do this. You would instead `return (ret)` at the end of the function. I don't have a preference either way but if you wanted to reduce the number of lines then that would be an option.